src/libcore/sync/atomic.rs

```diff
@@ -384,9 +419,11 @@ impl AtomicBool {
     /// was updated.
     ///
     /// `compare_and_swap` also takes an [`Ordering`] argument which describes the memory
-    /// ordering of this operation.
+    /// ordering of this operation. Notice that even when using [`AcqRel`], the operation
+    /// might fail and hence just perform an `Acquire` load, but not have `Release` semantics.
```

Comment:
This also applies to `compare_exchange` and `compare_exchange_weak`.

Reply:
No it does not. Those have a separate argument for the ordering used in case of failure.